### ISSUE TITLE
Style/AccessorGrouping false positive with multiple contexts

### ISSUE THREAD

**Issue by hlascelles** [rubocop-hq/rubocop#9428]
Consider the following code:

```ruby
module Foo
  class Bar
    attr_accessor :baz
  end

  class << self
    attr_accessor :bop
  end
end
```

The `attr_accessor` accessors are in different contexts, but rubocop sees them as the same:

```
=> foo.rb:3:5: C: Style/AccessorGrouping: Group together all attr_accessor attributes.
```

Rubocop version: 1.5.2. 

If this is a "good first issue to have a go at" I can try?

**Comment by koic** [comment]
This false positive cannot be reproduced. Can you try it with the latest RuboCop (1.18.1) ?

**Comment by hlascelles** [comment]
I do apologise, it was a delayed "cached" error in my IDE!
